connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedKafkaCluster.java

```diff
@@ -447,9 +326,9 @@ public void createTopic(String topic, int partitions, int replication, Map<Strin
      * @param adminClientConfig Additional admin client configuration settings.
      */
     public void createTopic(String topic, int partitions, int replication, Map<String, String> topicConfig, Map<String, Object> adminClientConfig) {
-        if (replication > brokers.length) {
+        if (replication > cluster.brokers().size()) {
```

Comment:
This logic is basically unchanged from the current state of things; the only reason it shows up in the diff is because there's a slightly different internal API for counting the number of brokers. I'm not sure why we had it originally, but unless it would leave to a regression I'd prefer not to block this PR on figuring that out and save it for a follow-up. Is that alright?

Reply:
> unless it would leave to a regression I'd prefer not to block this PR on figuring that out and save it for a follow-up. Is that alright?

make senses to me :)